frontend/src/containers/Subscription/SubscriptionFeed.js

```diff
@@ -20,6 +59,77 @@ class SubscriptionFeed extends Component {
     //     }
     // }
 
+    clickMoreButton = () => {
+        if (!this.props.subscriptionFinished) {
+            this.props.onGetSubscriptions({
+                page_number: this.props.subscriptionPageNum + 1,
+            })
+                .then(() => {
+                    const { subscriptions } = this.state;
+                    this.setState({
+                        subscriptions: subscriptions.concat(this.props.subscriptionItems),
+                    });
+                    this.clickMoreButtonNext();
+                }).catch(() => {});
+        } else {
+            this.clickMoreButtonNext();
+        }
+    }
+
+    clickMoreButtonNext = () => {
+        if (!this.props.recommendationFinished) {
+            this.props.onGetRecommendations({
+                page_number: this.props.recommendationPageNum + 1,
+            })
+                .then(() => {
+                    const { recommendations } = this.state;
+                    this.setState({
+                        recommendations: recommendations.concat(this.props.recommendationItems),
+                    });
+                    this.addRecoToSub();
+                }).catch(() => {});
+        } else {
+            this.addRecoToSub();
+        }
+    }
+
+    addRecoToSub = () => {
+        for (let i = this.state.recommendations.length - 1; i > this.state.start; i -= 1) {
+            const j = Math.floor(Math.random() * (i + 1));
+            [this.state.recommendations[i], this.state.recommendations[j]] = [
```

Comment:
I'm asking you because I don't know. Is it safe that changing an array of state directly without `setState`?

Reply:
Well it works well, but for safety, it would be better to assign other variables.